signals/signals.go

```diff
@@ -42,21 +40,23 @@ func handleSignals(finalizer func(), finishedChan chan struct{}) {
 	signal.Notify(sigChan, syscall.SIGTERM, syscall.SIGINT, syscall.SIGKILL)
 	defer signal.Stop(sigChan)
 	if finalizer != nil {
-		select {
-		case sig := <-sigChan:
-			// first signal use finalizer to terminate cleanly
+		switch sig := <-sigChan; sig {
```

Comment:
Nice! Thanks for looking into this and confirming this works as expected. It's cleaner to just have the `switch` instead a of `switch` nested in a single-case `select` 🥇